packages/react-bootstrap-table2/test/props-resolver/index.test.js

```diff
@@ -99,48 +100,292 @@ describe('TableResolver', () => {
         expect(cellEdit.cidx).toEqual(cidx);
       });
     });
+
+    describe('if cellEdit prop defined', () => {
+      const expectNonEditableRows = [1, 2];
+      const cellEdit = {
+        mode: Const.DBCLICK_TO_CELL_EDIT,
+        onEditing: sinon.stub(),
+        blurToSave: true,
+        beforeSaveCell: sinon.stub(),
+        afterSaveCell: sinon.stub(),
+        nonEditableRows: sinon.stub().returns(expectNonEditableRows)
+      };
+
+      beforeEach(() => {
+        const mockElement = React.createElement(BootstrapTableMock, {
+          data, keyField, columns, cellEdit
+        }, null);
+        wrapper = shallow(mockElement);
+      });
+
+      it('should resolve a cellEdit correctly', () => {
+        const cellEditInfo = wrapper.instance().resolveCellEditProps();
+        expect(cellEditInfo).toBeDefined();
+        expect(cellEditInfo.ridx).toBeNull();
+        expect(cellEditInfo.cidx).toBeNull();
+        expect(cellEditInfo.mode).toEqual(cellEdit.mode);
+        expect(cellEditInfo.onEditing).toEqual(cellEdit.onEditing);
+        expect(cellEditInfo.blurToSave).toEqual(cellEdit.blurToSave);
+        expect(cellEditInfo.beforeSaveCell).toEqual(cellEdit.beforeSaveCell);
+        expect(cellEditInfo.afterSaveCell).toEqual(cellEdit.afterSaveCell);
+        expect(cellEditInfo.nonEditableRows).toEqual(expectNonEditableRows);
+      });
+
+      it('should attach options to cellEdit props', () => {
+        const something = {
+          test: 1,
+          cb: sinon.stub()
+        };
+        const cellEditInfo = wrapper.instance().resolveCellEditProps(something);
+        expect(cellEditInfo).toBeDefined();
+        expect(cellEditInfo.test).toEqual(something.test);
+        expect(cellEditInfo.cb).toEqual(something.cb);
+      });
+    });
   });
 
-  describe('if cellEdit prop defined', () => {
-    const expectNonEditableRows = [1, 2];
-    const cellEdit = {
-      mode: Const.DBCLICK_TO_CELL_EDIT,
-      onEditing: sinon.stub(),
-      blurToSave: true,
-      beforeSaveCell: sinon.stub(),
-      afterSaveCell: sinon.stub(),
-      nonEditableRows: sinon.stub().returns(expectNonEditableRows)
-    };
+  describe('resolveCellSelectionProps', () => {
+    let cellSelectionInfo;
+    let selectRow;
+
+    describe('if selectRow was not defined', () => {
```

Comment:
Nice catch, if you have time, please help me write a **`cellEdit` not defined** test case, thanks a lots

Reply:
No problem

Reply:
It confused me, too. You've already written in [#index.test.js](https://github.com/react-bootstrap-table/react-bootstrap-table2/blob/development/row-selection/packages/react-bootstrap-table2/test/props-resolver/index.test.js#L76) so what I should do for you?

Reply:
nothing.